pkg/recipes/terraform/config/config.go

```diff
@@ -52,6 +60,84 @@ func New(moduleName string, envRecipe *recipes.EnvironmentDefinition, resourceRe
 	}
 }
 
+// getModuleSource is called to get the module source path with credential information
+// if template path represent a private source else returns the provided template path
+func getModuleSource(ctx context.Context, envConfig *recipes.Configuration, templatePath string, armOptions *arm.ClientOptions) (string, error) {
+	var source string
+	var err error
+	if strings.HasPrefix(templatePath, "git::") {
```

Comment:
Are we only supporting private registries that are accessed over HTTPS? Terraform supports other Git URLS (just looking at this: https://developer.hashicorp.com/terraform/language/modules/sources#github) so I'm wondering what the expected behavior would be if a user passed one of these formats

Reply:
thats correct, today we don't support private registries that are accessed over ssh. And for private git repository we expect user to provide the source in generic git format so all git platforms are supported.